Sources/Sentry/SentryInstallation.m

```diff
@@ -5,38 +5,40 @@
 
 @implementation SentryInstallation
 
-static NSString *volatile installationString;
++ (NSMutableDictionary*)installationStringsByCacheDirectoryPaths
+{
+    static dispatch_once_t once;
+    static NSMutableDictionary* dictionary;
+    
+    dispatch_once(&once, ^{
+        dictionary = [[NSMutableDictionary alloc] init];
+    });
+    return dictionary;
+}
 
-+ (NSString *)id
++ (NSString *)idWithCacheDirectoryPath:(NSString *)cacheDirectoryPath
```

Comment:
`m`: I know there are no tests for this, but please add some for the new logic with different cacheDirPaths. Please write them in Swift.